Assets/Scripts/CTF/FlagManager.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+using UnityEngine.InputSystem.Controls;
+
+public class FlagManager : MonoBehaviour
+{
+    public GameObject m_FlagIconToggler;
+    private FlagIconToggler m_FlagToggle;
+    public GameObject m_Flag;
+    private bool m_IsFlagInPosession;
+    // Start is called before the first frame update
+    void Start()
+    {
+        AetherInput.GetPlayerActions().Fire.performed += HandleFlag;
+        m_FlagToggle = m_FlagIconToggler.GetComponent<FlagIconToggler>();
+        Debug.Log(m_FlagToggle == null);
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (m_IsFlagInPosession)
+        {
+            m_FlagToggle.SetIcon();
+        }
+    }
+
+    void HandleFlag(InputAction.CallbackContext ctx)
+    {
+        ButtonControl button = ctx.control as ButtonControl;
+        if (!button.wasPressedThisFrame)
+            return;
+
+        if (!m_IsFlagInPosession)
+            return;
```

Comment:
Not a big deal, but your if statements have different styles here. Maybe stick with the no brace one for one-liners.